### PR TITLE
build: Migrate to Github Actions

### PR DESCRIPTION
**Pull request by simster7**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Comment by simster7** [comment]
@alexec Ready for an initial review. Will need to disable the required old Circle checks once ready.

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      - 'master'
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        test: ["smoke", "test", "test-e2e", "test-e2e-cron"]
```
Comment (alexec) [inline .github/workflows/ci-build.yaml:19]: nice! I think we can lose smoke if we have test-e2e

**Comment by alexec** [comment]
Looks good so far. Proof is in the pudding.

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (simster7) [reply .github/workflows/ci-build.yaml]: Removed `smoke`

**Pull request merged**